packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6596,6 +6698,67 @@ class MultiChildRenderObjectElement extends RenderObjectElement {
   }
 }
 
+/// A [RenderObjectElement] used to manage the root of a render tree.
+///
+/// Unlike any other render object element this element does not attempt to
+/// attach its [renderObject] to the closest ancestor [RenderObjectElement].
+/// Instead, subclasses must override [attachRenderObject] and
+/// [detachRenderObject] to attach/detach the [renderObject] to whatever
+/// instance manages the render tree (e.g. by assigning it to
+/// [PipelineOwner.rootNode]).
+abstract class RenderTreeRootElement extends RenderObjectElement {
+  /// Creates an element that uses the given widget as its configuration.
+  RenderTreeRootElement(super.widget);
+
+  @override
+  @mustCallSuper
+  void attachRenderObject(Object? newSlot) {
+    _slot = newSlot;
+    assert(_debugCheckMustNotAttachRenderObjectToAncestor());
+  }
+
+  @override
+  @mustCallSuper
+  void detachRenderObject() {
+    _slot = null;
+  }
+
+  @override
+  void updateSlot(Object? newSlot) {
+    super.updateSlot(newSlot);
+    assert(_debugCheckMustNotAttachRenderObjectToAncestor());
+  }
+
+  bool _debugCheckMustNotAttachRenderObjectToAncestor() {
+    if (!kDebugMode) {
+      return true;
+    }
+    if (_findAncestorRenderObjectElement() != null) {
+      throw FlutterError.fromParts(
+        <DiagnosticsNode>[
+          ErrorSummary(
+            'The RenderObject for ${toStringShort()} cannot maintain an independent render tree at its current location.',
+          ),
+          ErrorDescription(
+            'The ownership chain for the RenderObject in question was:\n  ${debugGetCreatorChain(10)}',
+          ),
+          ErrorDescription(
+            'This RenderObject is the root of an independent render tree and it cannot '
+            'attach itself to an ancestor in an existing tree. The ancestor RenderObject, '
+            'however, expects that a child will be attached.',
+          ),
+          ErrorHint(
+            'Try moving the subtree that contains the ${toStringShort()} widget into the '
+            'view property of a ViewAnchor widget or to the root of the widget tree, where '
+            'it is not expected to attach its RenderObject to its ancestor.',
```

Comment:
```suggestion
            'it is not expected to attach its RenderObject to its parent.',
```

It is the parent render object, right, not an arbitrary ancestor?